src/test/scala/com/sksamuel/scapegoat/inspections/empty/EmptyIfBlockTest.scala

```diff
@@ -28,5 +29,26 @@ class EmptyIfBlockTest extends InspectionTest {
       compileCodeSnippet(code)
       compiler.scapegoat.feedback.warnings.size shouldBe 2
     }
+
+    "should not report warning" - {
+      "the if-block is the return value" in {
+        val code =
+          """
+            |import scala.concurrent.Future
+            |import scala.concurrent.ExecutionContext.Implicits.global
+            |object Test {
+            |  Future("test").map(value => {
+            |   if (value.contains("foo")) {
```

Comment:
Don't have my IDE here, but I think the original authors' intent of scapegoat was to write:
```
Future("test").toOption match {
  case Some("foo") => ()
  case _ => throw new IllegalStateException("Bar!")
}
```
Does it make sense?

Reply:
Not entirely sure I follow, I based the sample on a further simplified version of the one in the issue:
```
  def checkFoo(things: Future[Seq[String]]): Future[Unit] = {
    things.map(ts => if (ts.contains("foo")) () else throw new Exception("No foo!"))
  }
```
The comment with the pattern matching is a valid workaround for this solution (although I'd prefer the transformWith solution which doesn't use flow control mechanisms), but does it really solve the problem scapegoat highlights? Instead of an empty if branch you now have an empty case branch. While empty case branches are maybe a bit more common to see, I wouldn't really see it as much different.
So I tend to agree with the the original issue and consider it a false-positive for cases where the empty branch (if or case in that regard) are being captured by the return type.

Reply:
> does it really solve the problem scapegoat highlights

I think the intended thinking was that empty branches in `ifs` are not OK, but empty case is fine.
Nevertheless, I don't feel strongly about that.

Reply:
if @sksamuel speaks up soon, I'd let him decide.
Otherwise I am fine with merging that.

Reply:
I have no strong opinion either so we can merge.